File: polyglot/windows/data/seis_windows_platform.r
seis_windows_platform <- list(
  id = "windows-r-analytics",
  languages = c("R", "SQL", "Python", "PowerShell"),
  quality_gates = c(
    "r_syntax_when_available",
    "windows_path_safety",
    "permission_scope",
    "offline_fallback",
    "event_log_awareness"
  ),
  offline_helper = TRUE,
  remote_bridge = TRUE
)

seis_windows_ready <- function(capability) {
  length(capability$languages) >= 4 &&
    length(capability$quality_gates) >= 5 &&
    capability$offline_helper &&
    capability$remote_bridge
}

File: polyglot/windows/native/seis_windows_platform.cpp
#include "seis_windows_platform.hpp"

namespace seis::windows {

Capability platform_capability() {
  return Capability{
      "windows-native-cpp",
      {"C++", "C", "PowerShell", "Python"},
      {"native_cpp_syntax_when_available",
       "windows_path_safety",
       "permission_scope",
       "offline_fallback",
       "event_log_awareness"},
      true,
      true};
}

bool is_ready_for_seis_agent(const Capability& capability) {
  return capability.languages.size() >= 4 &&
         capability.quality_gates.size() >= 5 &&
         capability.offline_helper &&
         capability.remote_bridge;
}

}  // namespace seis::windows
